docs/testing.md

```diff
@@ -1,26 +1,66 @@
 Testing
 =======
 
-Running `make test` runs most of the integrated tests. You should have
-`valgrind` installed.
+The tests in libvfio-user are organized into a number of suites
 
-Running `make pre-push` runs the above builds and tests in different configurations: GCC,
-clang, and with ASAN enabled.
+* native - old tests that use native binaries written in C
+* python - modern tests written in python
+* valgrind - tests that can run under valgrind (only enabled
+             if `-Dvalgrind=true` to be passed when `meson`
+	     was first invoked)
 
-There are some [older unit tests](test/unit-tests.c) written in C, but most
-tests are now done via Python, in the [test/py](test/py) sub-directory. You can
-run just the Python tests via `make pytest` or `make pytest-valgrind`.
+Running `meson test -C build` runs all the suites:
+
+```
+meson build -Dvalgrind=true
+meson test -C build
+```
+
+It is possible to be selective about which tests are run
+by including or excluding suites:
+
+```
+meson test -C build --suite=python
+meson test -C build --no-suite=valgrind
+```
+
+To run with ASAN enabled, pass the `-Db_sanitize=address`
+option to meson. Note, this is incompatible with enabling
+valgrind
+
+```
+meson build -Db_sanitize=address
+meson test -C build
+```
+
+The `.github/workflows/pull_request.sh` script run a
+sequence of builds in various configurations. This is
+invoked for all pull requests, but can be launched
+manually by contributors ahead of opening a pull
+request.
 
 The master branch is run through [Coverity](scan.coverity.com) when a new PR
 lands.
 
-You can also run `make gcov` to get code coverage reports.
+Coverage reports can be enabled via meson
+
+```
+meson build -Db_coverage=true
```

Comment:
I had to install gcovr (OK), but I can't find out how to get the detailed annotated output with meson? It only gives me summaries.

Reply:
OK, I kind of prefer the old way I could load into vim, but this will do indeed, thanks.